packages/react-native/ReactAndroid/src/main/java/com/facebook/react/views/text/frescosupport/FrescoBasedReactTextInlineImageSpan.kt

```diff
@@ -0,0 +1,160 @@
+/*
+* Copyright (c) Meta Platforms, Inc. and affiliates.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+package com.facebook.react.views.text.frescosupport
+
+import android.content.res.Resources
+import android.graphics.Canvas
+import android.graphics.Paint
+import android.graphics.PorterDuff
+import android.graphics.drawable.Drawable
+import android.net.Uri
+import android.widget.TextView
+import com.facebook.drawee.controller.AbstractDraweeControllerBuilder
+import com.facebook.drawee.generic.GenericDraweeHierarchy
+import com.facebook.drawee.generic.GenericDraweeHierarchyBuilder
+import com.facebook.drawee.view.DraweeHolder
+import com.facebook.imagepipeline.request.ImageRequest
+import com.facebook.imagepipeline.request.ImageRequestBuilder
+import com.facebook.react.bridge.ReadableMap
+import com.facebook.react.modules.fresco.ReactNetworkImageRequest
+import com.facebook.react.uimanager.PixelUtil
+import com.facebook.react.views.image.ImageResizeMode
+import com.facebook.react.views.text.internal.span.TextInlineImageSpan
+
+/**
+ * FrescoBasedTextInlineImageSpan is a span for Images that are inside <Text/>. It computes its size
+ * based on the input size. When it is time to draw, it will use the Fresco framework to get the
+ * right Drawable and let that draw.
+ *
+ * <p>Since Fresco needs to callback to the TextView that contains this, in the ViewManager, you
+ * must tell the Span about the TextView
+ *
+ * <p>Note: It borrows code from DynamicDrawableSpan and if that code updates how it computes size
+ * or draws, we need to update this as well.
+ */
+public class FrescoBasedReactTextInlineImageSpan(
+    resources: Resources,
+    height: Int,
+    width: Int,
+    private val tintColor: Int,
+    uri: Uri?,
+    private val headers: ReadableMap?,
+    private val draweeControllerBuilder: AbstractDraweeControllerBuilder<*, *, ImageRequest, *>,
```

Comment:
Try:

```suggestion
    private val draweeControllerBuilder: AbstractDraweeControllerBuilder<*, ImageRequest, *, *>,
```

Was a bit tricky to understand as well, but this worked

Reply:
Thank you for this! It worked. My bad I was passing ImageRequest as the third param while the class signature expects it to be on the 2nd:

https://github.com/facebook/fresco/blob/b79964050f2bd82c8e6e9bc5a96dcfe61f95ae67/drawee/src/main/java/com/facebook/drawee/controller/AbstractDraweeControllerBuilder.java#L36